### ISSUE TITLE
AttributeError: 'BlockBlobService' object has no attribute 'get_blob_client'

### ISSUE THREAD

**Issue by himachida** [Azure/azure-cli#22240]
## Describe the bug
When use "az storage blob delete-batch" with an account key, the following error occures.

**Command Name**
`az storage blob delete-batch`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
'BlockBlobService' object has no attribute 'get_blob_client'
Traceback (most recent call last):
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 231, in invoke
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 658, in execute
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 721, in _run_jobs_serially
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 713, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/__init__.py", line 409, in new_handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/__init__.py", line 408, in new_handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 692, in _run_job
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 328, in __call__
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/operations/blob.py", line 709, in storage_blob_delete_batch
  File "D:\a\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/util.py", line 33, in collect_blob_objects
AttributeError: 'BlockBlobService' object has no attribute 'get_blob_client'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az storage blob delete-batch --source {} --account-name {} --account-key {} --pattern {}`

## Environment Summary
```
Windows-10-10.0.22000-SP0
Python 3.10.4
Installer: MSI

azure-cli 2.36.0

Extensions:
account 0.2.1

Dependencies:
msal 1.17.0
azure-mgmt-resource 20.0.0
```

**Comment by yonzhan** [comment]
storage

**Comment by taiyokato** [comment]
Tested this at 2.30, 2.35, and 2.36.
Worked fine prior to 2.36.
2.36 works too, only if passed a regex to `--pattern` (non-regex file name breaks)

Perhaps the the `blob_service` in this commit is supposed to be `BaseBlobService` instead of `BlobServiceClient` class?
https://github.com/Azure/azure-cli/pull/22072/commits/39ba7483282f2e33b98473997ac2d872ab9d7cd1

https://github.com/evelyn-ys/azure-cli/blob/39ba7483282f2e33b98473997ac2d872ab9d7cd1/src/azure-cli/azure/cli/command_modules/storage/util.py#L33

**Comment by Abinaya587** [comment]
Still getting below error,when using az storage blob delete-batch --source {} --account-name {} --account-key {} --pattern {}
AttributeError: 'BlockBlobService' object has no attribute 'get_blob_client'

**Comment by calvinhzy** [comment]
It should come as part of the release on 05/24/22.

**Comment by calvinhzy** [comment]
It should be part of the fix in version 2.37.0.
